Frameworks/CoreText/DWriteWrapper_CoreText.mm

```diff
@@ -802,4 +803,140 @@ CFStringRef _DWriteFontCopyName(const ComPtr<IDWriteFontFace>& fontFace, CFStrin
     }
 
     return _DWriteFontCopyInformationalString(fontFace, informationalStringId);
+}
+
+/**
+ * Custom IDWriteGeometrySink class that captures the glyph runs generated by
+ * DWrite for the given TextLayout and TextRenderer constraints.
+ *
+ * Notes:
+ *
+ * DWrite provides negative values for y-coordinates, whereas CGPath expects positive ones
+ * As such, functions in this class will invert y-coordinates when passing points from DWrite to CG
+ *
+ * IDWriteFontFace::GetGlyphRunOutline uses this class in a single-threaded manner (dumping the draw instructions from the font linearly)
+ * As such, this class is deliberately left thread-unsafe.
+ */
+class CGPathGeometrySink : public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IDWriteGeometrySink> {
+protected:
+    InspectableClass(L"Windows.Bridge.DirectWrite", TrustLevel::BaseTrust);
+
+public:
+    CGPathGeometrySink(const CGAffineTransform* transform);
+
+    HRESULT RuntimeClassInitialize();
+
+    void STDMETHODCALLTYPE AddBeziers(_In_ const D2D1_BEZIER_SEGMENT* beziers, unsigned int beziersCount) {
+        // Note that both AddBeziers and CGPathAddCurveToPoint use cubic Beziers (rather than quadratic)
+        for (unsigned int i = 0; i < beziersCount; ++i) {
+            CGPathAddCurveToPoint(_cgPath.get(),
+                                  _transform,
+                                  beziers[i].point1.x,
+                                  -beziers[i].point1.y,
```

Comment:
>-beziers[i].point1.y [](start = 34, length = 20)

nit: maybe add parens around this so the unary - sticks out more, right now it's super sneaky